pandas/tests/io/test_sql.py

```diff
@@ -2698,6 +2700,64 @@ def test_drop_table(conn, request):
         assert not insp.has_table("temp_frame")
 
 
+@pytest.mark.parametrize("conn", all_connectable)
+def test_delete_rows_success(conn, test_frame1, request):
+    table_name = "temp_frame"
+    conn = request.getfixturevalue(conn)
+    pandasSQL = pandasSQL_builder(conn)
+
+    with pandasSQL.run_transaction():
+        assert pandasSQL.to_sql(test_frame1, table_name) == test_frame1.shape[0]
+
+    with pandasSQL.run_transaction():
+        assert pandasSQL.delete_rows(table_name) is None
+
+    assert count_rows(conn, table_name) == 0
+    assert pandasSQL.has_table("temp_frame")
+
+
+@pytest.mark.parametrize("conn", all_connectable)
+def test_delete_rows_is_atomic(conn, request):
+    import adbc_driver_manager
```

Comment:
Need to be careful about how we structure the tests. The `import adbc_driver_manager` assumes that is installed, but this is an optional component.

I think you can just do `if "adbc" in conn:` to get the same effect but avoid the hard import requirement, so that the tests don't fail where this library is not installed.

Ditto for SQLAlchemy on the next line

Reply:
Humm...we can remove both imports but that would result in `with pytest.raises(Exception):` on line 2749. Not the most readable but the exception per se is not relevant to the test... Any exception triggered during the transaction should fail...
WDYT of replacing it in favor of `Exception` ?

Reply:
You can structure it like:

```python
if "adbc" in conn:
    mod = adbc_driver_manager
    exc = mod.TheErrorType
else:
    ...
```

So no need to catch a bare Exception

This does also bring up an interesting question if we should have a more generic pd.SQLError to catch these and give the users one error surface, but let's leave that to a separate initiative

Reply:
Sorry make that:

```python
mod = pytest.importorskip("adbc_driver_manager")
```

you will see this pattern a few other places in this module already if you need a reference

Reply:
Perfect 👌 . Just did it !